### ISSUE TITLE
Connector v0.25.0 does not seem to support columns descriptions

### ISSUE THREAD

**Issue by AlexisBRENON** [GoogleCloudDataproc/spark-bigquery-connector#660]
Hi,

Starting with latest version of the connector my Spark job failed with the following error:
```
Exception in thread "main" java.lang.RuntimeException: Failed to write to BigQuery
	at com.google.cloud.spark.bigquery.BigQueryWriteHelper.writeDataFrameToBigQuery(BigQueryWriteHelper.scala:70)
	at com.google.cloud.spark.bigquery.BigQueryInsertableRelation.insert(BigQueryInsertableRelation.scala:43)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelation(BigQueryRelationProvider.scala:111)
       [...]
Caused by: java.lang.IllegalArgumentException: com.google.cloud.bigquery.connector.common.BigQueryConnectorException$InvalidSchemaException: Destination table's schema is not compatible with dataframe's schema
	at com.google.cloud.spark.bigquery.repackaged.com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at com.google.cloud.bigquery.connector.common.BigQueryClient.loadDataIntoTable(BigQueryClient.java:466)
	at com.google.cloud.spark.bigquery.BigQueryWriteHelper.loadDataToBigQuery(BigQueryWriteHelper.scala:92)
	at com.google.cloud.spark.bigquery.BigQueryWriteHelper.writeDataFrameToBigQuery(BigQueryWriteHelper.scala:67)
	... 40 more
```

After some investigation I finally tried to read data from the table and compare the resulting schema with the one generated in Scala.
Schemas differ on some fields, the ones that contain description.
Description seems to be put in the `metadata` field of the `StructField` class, but it isn't to add such metadata to schema inferred from case classes. Hence, the schema equality check fails.

I didn't try to run the tests in the repo because it requires too much setup for the moment but something like this should do the job.
Given a BQ table:
```sql
CREATE TABLE `test_dataset.table_with_description`
(
  str STRING OPTIONS(description="A description"),
);
```
Then, the following test should pass.
```scala
case class Foo(str: Option[String])

@Test
def testIndirectWriteToBigQueryWithDiffInSchemaNullableFieldAndDisableModeCheck(): Unit = {
  assumeThat(writeMethod, equalTo(SparkBigQueryConfig.WriteMethod.INDIRECT))
  spark.conf().set("temporaryGcsBucket", temporaryGcsBucket)
  val ds = spark.createDataset(Seq(Foo("string")))

  ds.write()
      .format("bigquery")
      .mode(SaveMode.Append)
      .option("writeMethod", writeMethod.toString())
      .save("test_dataset.table_with_description");

  val query = "select * from test_dataset.table_with_description"
  val numOfRows: Int = bq.query(QueryJobConfiguration.of(query)).getTotalRows()
  assertThat(numOfRows).isEqualTo(1);
}
```

**Comment by suryasoma** [comment]
Hello @AlexisBRENON, the issue is fixed and the fix would be available in the next release.

**Comment by suryasoma** [comment]
Hey @AlexisBRENON, please find the fix in the latest release 0.25.2. Thanks
